data-processing/common/commands/fetch_s2_data.py

```diff
@@ -99,13 +102,28 @@ def process(self, item: ArxivId) -> Iterator[S2Metadata]:
                 data = resp.json()
                 references = []
                 for reference_data in data["references"]:
+
+                    if reference_data["paperId"] is None:
+                        # can be legit references not in our corpus, or bad references from bogus extractions
+                        continue
+
+                    external_ids = reference_data["externalIds"]
+                    if external_ids:
+                        corpus_id = external_ids.get("CorpusId")  # always present (scholar/citations/MatchResult.scala)
```

Comment:
Looking back at this comment... I wonder if it might have more potential to confuse someone than to help, because of all the intermediate steps between citation matching and what the public API returns, as well as the fact that this is an open source library so not everyone will have access to the code you're pointing at... What do you think? Maybe we could just say every canonical paper has a corpus id, so you shouldn't be able to have a matched reference without one?

Reply:
oh yeah!  lol I forgot scholarphi is open source.  Yes, I will change the comment to `# every canonical paper will always have a CorpusId`